master/internal/trial.go

```diff
@@ -1030,8 +1030,13 @@ func classifyStatus(state terminatedContainerWithState) aproto.ContainerStopped
 }
 
 func (t *trial) reset() error {
-	step := t.sequencer.RollBackSequencer()
-	if err := t.db.RollBackTrial(t.id, step); err != nil {
+	t.sequencer.RollBackSequencer()
+	w, err := t.sequencer.Workload()
+	if err != nil {
+		return errors.Wrap(err, "failed to get workload from sequencer in reset")
+	}
+	err = t.db.RollBackTrial(t.id, w.TotalBatches())
```

Comment:
```suggestion
	w, err := t.sequencer.Workload()
	if err != nil {
		return errors.Wrap(err, "failed to get workload from sequencer in reset")
	}
	err = t.db.RollBackTrial(t.id, w.TotalBatchesProcessed)
```
This used to take the last step we completed, now it takes the current step we're on (just in batches instead of steps) which I think introduces a master restart bug. This means we're just to miss rolling back the current step we're on since `t.db.RollBackTrial` purges everything _after_ `w.TotalBatches()`, not including it, and the trial is going to start and run the batch at `w.TotalBatches`.

Should able to confirm this is broken/fixed by this change by running the adaptive chaos noop and looking at the logs.

Reply:
This used to take the last step we completed, now it takes the current step we're on (just in batches instead of steps) which I think introduces a master restart bug.

> Good catch!